src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -110,16 +156,23 @@ public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> customParseDate(String date) {
-        return Optional.empty();
+        if (Integer.parseInt(date.substring(0, 2)) > 31) {
+            return Optional.empty();
+        }
```

Comment:
if you'll get unexisting month? better
```suggestion
final String localDatePattern = "d MMM yyyy";
        try {
            return Optional.of(LocalDate.parse(date, DateTimeFormatter.ofPattern(localDatePattern)
                    .withLocale(new Locale("en"))));
        } catch (DateTimeParseException e) {
            return Optional.empty();
        }
```

Reply:
done, but in the method withLocale i gave enum